login-utils/login.c

```diff
@@ -1365,8 +1367,11 @@ static void initialize(int argc, char **argv, struct login_context *cxt)
 			*p++ = ' ';
 #endif
 	}
-
+#ifdef HAVE_CLOSE_RANGE
+	close_range(3, ~0U);
+#else
 	close_all_fds(wanted_fds, ARRAY_SIZE(wanted_fds));
+#endif
 }
```

Comment:
It seems our `close_all_fds()` is used by login.c only (+ test in fileutils.c). 

What about to change it to have the same arguments as close_range(), it means  close_all_fds(first, last). Then we can remove `wanted_fds[]` at all from code and keep all the #ifdefs in fileutils.c.

 It would be probably also nice to use STDERR_FILENO+1 rather than hardcoded `3` to make it more readable.

Reply:
Really good idea. I changed the `close_all_fds()` to have same interface as `close_range()` so that moving in between old before and after system call availability is smooth, see a75700d86e2a0245d28fdf2762638d07efc549a0. Perhaps in around five to ten years the fallback can be removed. We'll see.